### PR TITLE
Githubactions docker buildx

### PR DESCRIPTION
**Pull request by rnsc**
This PR aims at fixing the following issue: https://github.com/j8r/dockerfiles/issues/20
It's a bit more complex than the simple trigger of the builds on Docker Hub, but it adds multi arch images.

The only thing needed for this to work is the provisioning of two secrets:
DOCKER_HUB_USERNAME
DOCKER_HUB_TOKEN

@j8r I hope you'll like this approach.

### CODE REVIEW

**Comment by j8r** [comment]
Great work, thanks @rnsc. Can this 3 jobs be in a single file, to avoid repetition (merged matrix)?

This is also a welcome change because [autobuilds will be now disabled](https://www.docker.com/blog/changes-to-docker-hub-autobuilds/). I'll be able to remove the Docker Hub integration altogether.

**Comment by rnsc** [comment]
Sure, I did split them for clarity and ease of use (only triggering build when the the files for a specific OS were changed).

I'm not against merging everything into one matrix.
I'll update the PR.

**Comment by rnsc** [comment]
I'll also switch the tags from my personal repo to jre

**Comment by rnsc** [comment]
@j8r all done :)

**Comment by rnsc** [comment]
Btw Debian 8 doesn't have arm images, so it only builds an amd64 image.
Centos8 doesn't have an arm/v7 image, so it's also removed.
And there's no arm/v8 because the buildx doesn't support it.

**Comment by j8r** [comment]
That's because arm64 is at least armv8. They could have also called `armv7` `armhf`.

**Comment by j8r** [comment]
> only triggering build when the files for a specific OS were changed

It is possible? Generally all the CIs will run whatever the change in the repo is.

**Comment by rnsc** [comment]
> > only triggering build when the files for a specific OS were changed
> 
> 
> 
> It is possible? Generally all the CIs will run whatever the change in the repo is.

Yes, if you look at the top here

https://github.com/j8r/dockerfiles/commit/63bf17e310450a752a561d94c534126bb031f49d

There's a path defined, so it would only trigger if of the files in that was updated. That's when having multiple build file is either to maintain imho.
Let me know what you think and I can rollback to the split version.

**Comment by j8r** [comment]
I guess it's fine with the merged version, the repository has very few changes. Plus, this does not prevent to rebuild other versions of a given distribution - not really worth it. Plus, this Dockerfiles are very simple and quick to build.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,21 +7,131 @@ on:
...
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
```
Comment (j8r) [inline .github/workflows/build.yml:123]:
```suggestion
          username: jrei
```
I've seen this in the official example, but no need: not a secret.

Comment (rnsc) [reply .github/workflows/build.yml]: Ok, fixed :)

**Comment by j8r** [comment]
Thanks a lot @rnsc!

**Pull request merged**